streams/src/main/java/org/apache/kafka/streams/state/internals/CachingWindowStore.java

```diff
@@ -156,6 +156,19 @@ public synchronized void put(final Bytes key, final byte[] value, final long tim
         cache.put(name, cacheFunction.cacheKey(keyBytes), entry);
     }
 
+    @Override
+    public byte[] fetch(final Bytes key, final long timestamp) {
+        validateStoreOpen();
+        final Bytes bytesKey = WindowStoreUtils.toBinaryKey(key, timestamp, 0, bytesSerdes);
```

Comment:
`toBinaryKey(key.get(), timestamp, 0);`

Should be fixes in L153 above, too. (maybe some other places as well)

Reply:
Doesn't the method the `bytesSerdes` to extract the underlying `byte[]` array?

Reply:
key.get() returns the `byte[]` array directly -- no need to call `byteSerdes.serialize()` that does call `key.get()` internally -- this is old code that was written before we remove the generics for stores and forces them to be of type `<Bytes,byte[]>` -- thus, we can simplify some code now.

Reply:
Ack.

Bytes serde does not do anything but just read the underlying bytes array from `Bytes` key. By doing it another way we can save the extra func calls.